tree/30_generic_methods/package_state_options.cf

```diff
@@ -40,23 +40,32 @@
 bundle agent package_state_options(name, version, architecture, provider, state, options)
 {
   vars:
-      "class_prefix"                string => "package_state_options_${name}";
+    any::
+      "class_prefix"          string => "package_state_options_${name}";
+      "args"                  slist  => { "${name}", "${version}", "${architecture}", "${provider}", "${state}", "${options}" };
+      "report_param"          string => join("_", args);
+      "full_class_prefix"     string => canonify("package_state_options_${report_param}");
+      "old_class_prefix"      string => string_head("${full_class_prefix}", "1000");
```

Comment:
this is really confusing to set "old_class_prefix" to the previous class_previous while "class_prefix" is the "old_class_prefix"
Can you use "previous_class_prefix" ?